### PR TITLE
Add `Share` module

### PR DESCRIPTION
**Pull request by Trancever**
Add bindings for share module from Rebolt.

### CODE REVIEW

File: src/share.rei [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  (~title: string=?, ~message: string, ~url: string) => makeContent;
+let makeOptions:
+  (
+    ~subject: string=?,
+    ~tintColor: string=?,
+    ~excludedActivityTypes: array(string)=?,
+    ~dialogTitle: string=?
+  ) =>
+  makeOptions;
+let _share: (makeContent, makeOptions) => Js.Promise.t(bool);
```
Comment (wokalski) [inline src/share.rei]: This wasn't intended was it?

File: src/share.rei [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+let share:
+  (
+    ~content: content,
+    ~subject: string=?,
+    ~tintColor: string=?,
+    ~excludedActivityTypes: array(string)=?,
```
Comment (wokalski) [inline src/share.rei]: Are activity types a well defined set? If yes, maybe this could be a variant.

**Comment by grabbou** [comment]
Updated with better API. @wokalski, answering one of your requested changes, we cannot unfortunately type `excludedActivityTypes` prop as its meant to be literally an array of strings. You use it to define a list of `share providers` to disable when iOS is about to show its share menu (e.g. no `post to twitter`).

**Comment by wokalski** [comment]
👍

**Pull request merged**